continuous-integration/starter-kits/compile-release-publish/java-maven/simple/.ci/Dockerfile.buildDeploy

```diff
@@ -0,0 +1,32 @@
+# docker image rm slim:mavenbuilddeploy ; docker build --build-arg UID=$(id -u) --build-arg GID=$(id -g) --network=host --tag "slim:mavenbuilddeploy" --force-rm --no-cache --file ".ci/Dockerfile.buildDeploy" .
+# docker run --rm -it -v ${PWD}:${PWD} -v ${HOME}/.m2:$PWD/.m2 -v ${HOME}/.m2:/home/jenkins/.m2 -w ${PWD} --user "${UID_GID}" slim:mavenbuilddeploy
+FROM 3.6.3-openjdk-8
+
+ARG DEBIAN_FRONTEND=noninteractive
+# mapping custom ids should take place here
+ARG GID=1003
+ARG UID=1003
+ARG USR=jenkins
+ARG WORKSPACE="/home/${USR}"
+
+ENV CONTAINERIZED="true"
+ENV HOME="${WORKSPACE}"
+ENV PATH="${PATH}:."
+
+# https://issues.jenkins-ci.org/browse/JENKINS-47026
+RUN { getent group ${GID} || groupadd -f -g ${GID} ${USR} ; } && \
+    { getent passwd ${UID} || useradd -u ${UID} -s /bin/sh -d ${HOME} -g ${GID} ${USR} ; }
+
+# includes jq, python and testrail reporter
+RUN mkdir -p ${HOME} && chmod 777 ${HOME} && \
+    apt-get update && \
+    apt-get install -y --no-install-recommends \
+#            build-essential \
+            jq \
+            python3-wheel \
+            python3-dev \
+            python3-setuptools \
+            python3-pip && \
+    python3 -m pip install behave-testrail-reporter~=0.4.0
```

Comment:
See comment above about test rail.

Reply:
Dockerfile is generic but includes this tool to allow reporting should it be desired.

Reply:
I suppose the question is - are we mentioning / using TestRail in any other part of the starter kit? If its entirely optional and not mentioned in the guide, might just be extra baggage that people can add on their own. I think if we do bundle it, the starter kit should mention its use in some way.